cmake/utils.cmake

```diff
@@ -48,7 +48,7 @@ function(register_exe name srcs test)
     add_executable(${name} ${srcs})
     target_link_libraries(${name} ${LIB_NAME} ${EXTRA_SHARED_LIBS} ${ARGV3})
     if("x${test}" STREQUAL "xtest")
-        add_test(${name} ${name})
+        add_test(${name} ${QEMU_PATH} ${name})
```

Comment:
ctest (/usr/bin/ctest) itself is x64 executable and it runs the tests listed in CTestTestfile.cmake.
It seems to me that the only way to run the test programs on qemu is to modify CMake files.
If CI ran on docker environment with privileged option, CMakeLists.txt didn't need to be modified. We could apt-install qemu-user-binfmt (it requires privileged option). It automatically uses qemu-aarch64 for AArch64 executables  :-O.

Reply:
You are right, @kawakami-k! I completely missed the fact that ctest remains an x64 binary :)

While `qemu-user-binfmt` is definitely a convenient feature, it would be safer to not rely it (since some users may want to test Aarch64 build not through CI / docker). So I am fine with the current approach.

The only suggestion is to replace QEMU_PATH with something more generic and instead of changing `add_test` in all the places introduce a function `add_dnnl_test()` that would do that for the others.
Something like this: emfomenk/mkl-dnn@4a79bf1ec3087b1e692483ad3e7179bc2fee4fd1. If you are oK with this patch, feel free to cherry-pick it.

Reply:
Minor comment on the commit above: it doesn't cover `examples/CMakeLists.txt.in` as this file is standalone and is used in binary distributions (so probably no need to cover examples with emulator in this case). The examples that are part of the regular build are covered by `examples/CMakeLists.txt`.